api/src/main/scala/org/ensime/api/config.scala

```diff
@@ -29,60 +28,33 @@ final case class EnsimeConfig(
   /* Proposed alternatives to the legacy wire format field names */
   def root = rootDir
   val referenceSourceJars =
-    (javaSources ++ subprojects.flatMap(_.referenceSourceRoots)).toSet
+    (javaSources ++ projects.flatMap(_.librarySources)).toSet
 
   // some marshalling libs (e.g. spray-json) might not like extra vals
-  val modules = subprojects.map { module => (module.name, module) }.toMap
+  val modules = projects.map { module => (module.id, module) }.toMap
 
   def compileClasspath: Set[File] = modules.values.toSet.flatMap {
-    m: EnsimeModule => m.compileDeps ++ m.testDeps
+    m: EnsimeProject => m.libraryJars
```

Comment:
@VC1995 with the work you are doing in creating an Analyzer per module, I don't think we will need these any longer.
Same for `targetClasspath`, `allJars`, `allTargets` and `allDocJars`.